src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/HKDF.cs

```diff
@@ -84,10 +83,14 @@ private static void Extract(HashAlgorithmName hashAlgorithmName, int hashLength,
         public static byte[] Expand(HashAlgorithmName hashAlgorithmName, byte[] prk, int outputLength, byte[]? info = null)
         {
             ArgumentNullException.ThrowIfNull(prk);
-
             ArgumentOutOfRangeException.ThrowIfNegativeOrZero(outputLength);
 
-            int hashLength = HashLength(hashAlgorithmName);
+            int hashLength = Helpers.HashLength(hashAlgorithmName);
+
+            if (prk.Length < hashLength)
```

Comment:
This validation was in a "core" method that got abstracted, so it got pushed up in to the public API surface.